src/pushNotifier.ts

```diff
@@ -24,15 +24,29 @@ export class PushNotifier {
 		this.url = url;
 	}
 
-	notify({ title = "cross-seed", body, ...rest }: PushNotification): void {
+	async notify({
+		title = "cross-seed",
+		body,
+		...rest
+	}: PushNotification): Promise<void> {
 		if (this.url) {
-			fetch(this.url, {
-				method: "POST",
-				headers: { "Content-Type": "application/json" },
-				body: JSON.stringify({ title, body, ...rest }),
-			}).catch(() => {
-				logger.error({ message: "Failed to send push notification" });
-			});
+			try {
+				const response = await fetch(this.url, {
+					method: "POST",
+					headers: { "Content-Type": "application/json" },
+					body: JSON.stringify({ title, body, ...rest }),
+				});
+
+				if (!response.ok) {
+					logger.error({
+						message: `Failed to send push notification (Status Code: ${response.status} ${response.statusText})`,
+					});
+				}
+			} catch (error) {
+				logger.error({
+					message: `Failed to send push notification (Error: ${error})`,
```

Comment:
a convention I've developed is doing `logger.debug(error)` after `logger.error(whatever)`, but not putting an error object directly into a template string.

I realized far too late that Winston doesn't log stack traces by default, but it's extensible enough that we can implement a custom formatter that special cases on `Error` objects and also logs their stack trace. And once we do that, we can get this error's stack trace to log "for free" if we use the convention.